src/shared/configUtils.ts

```diff
@@ -35,27 +48,29 @@ export class ConfigUtils {
     return this.#globalConfig;
   }
 
-  public static async getOrCreateIdentityToken(): Promise<string> {
-    let token = await this.getIdentityToken();
-    if (!token) {
-      token = CryptoUtils.generateIdentityToken();
-      await this.writeIdentityToken(token);
+  public static async getOrCreateIdentityToken(username: string, connection: Connection): Promise<string> {
+    let identityData = await this.getIdentityData();
+    if (!identityData) {
+      const token = CryptoUtils.generateIdentityToken();
+      const entityId = await this.saveIdentityTokenToServer(token, connection);
+      identityData = new LocalWebServerIdentityData(token);
+      identityData.usernameToServerEntityIdMap[username] = entityId;
+      await this.writeIdentityData(identityData);
+      return token;
+    } else {
+      let entityId = identityData.usernameToServerEntityIdMap[username];
+      if (!entityId) {
+        entityId = await this.saveIdentityTokenToServer(identityData.identityToken, connection);
+        identityData.usernameToServerEntityIdMap[username] = entityId;
+        await this.writeIdentityData(identityData);
+      }
+      return identityData.identityToken;
     }
-    return token;
-  }
-
-  public static async getIdentityToken(): Promise<string | undefined> {
-    const config = await ConfigAggregator.create({ customConfigMeta: configMeta });
-    // Need to reload to make sure the values read are decrypted
-    await config.reload();
-    const identityToken = config.getPropertyValue(ConfigVars.LOCAL_WEB_SERVER_IDENTITY_TOKEN);
-
-    return identityToken as string;
   }
 
-  public static async writeIdentityToken(token: string): Promise<void> {
-    const config = await this.getConfig();
-    config.set(ConfigVars.LOCAL_WEB_SERVER_IDENTITY_TOKEN, token);
+  public static async writeIdentityData(identityData: LocalWebServerIdentityData): Promise<void> {
+    const config = await this.getGlobalConfig();
+    config.set(ConfigVars.LOCAL_WEB_SERVER_IDENTITY_DATA, JSON.stringify(identityData));
```

Comment:
This value can be JSON. We should just set the `identityData` value itself, not a parsed/stringified version of it.

Reply:
When I remove parse/stringify I am getting the following errors. Maybe we do want to keep them parsed/stringified?

```
src/shared/configUtils.ts:66:59 - error TS2345: Argument of type 'LocalWebServerIdentityData' is not assignable to parameter of type 'Optional<AnyJson>'.
  Type 'LocalWebServerIdentityData' is not assignable to type 'JsonMap'.
    Index signature for type 'string' is missing in type 'LocalWebServerIdentityData'.

66     config.set(ConfigVars.LOCAL_WEB_SERVER_IDENTITY_DATA, identityData);
```

```
error TS2352: Conversion of type 'string | number | boolean | JsonMap | JsonArray' to type 'LocalWebServerIdentityData' may be a mistake because neither type sufficiently overlaps with the other. If this was intentional, convert the expression to 'unknown' first.
  Type 'AnyJson[]' is missing the following properties from type 'LocalWebServerIdentityData': identityToken, usernameToServerEntityIdMap
```

Reply:
I think it might be because I have LocalWebServerIdentityData as class. I'm going to change it to `type`.

Reply:
Yup, that was it.